packages/compose/src/hooks/use-resize-observer/index.native.js

```diff
@@ -0,0 +1,56 @@
+/**
+ * External dependencies
+ */
+import { View, StyleSheet } from 'react-native';
+/**
+ * WordPress dependencies
+ */
+import { useState, useCallback } from '@wordpress/element';
+
+/**
+ * Hook which allows to listen the resize event of any target element when it changes sizes.
+ *
+ * @return {Object} Measurements object with properties `width` and `height`.
+ *
+ * @example
+ *
+ * ```js
+ * const App = () => {
+ * 	const [ resizeListener, sizes ] = useResizeObserver();
+ *
+ * 	return (
+ * 		<View>
+ * 			{ resizeListener }
+ * 			Your content here
+ * 		</View>
+ * 	);
+ * };
+ * ```
+ *
+ */
+const useResizeObserver = () => {
+	const [ measurements, setMeasurements ] = useState( null );
+
+	const onLayout = useCallback( ( { nativeEvent } ) => {
+		const { width, height } = nativeEvent.layout;
+		setMeasurements( ( prevState ) => {
+			if (
+				! prevState ||
+				prevState.width !== width ||
+				prevState.height !== height
+			) {
+				return { width, height };
+			}
+			return prevState;
+		} );
+	}, [] );
+
+	return [
+		// `key` is needless in that place since we are not rendering an array of elements
+		// eslint-disable-next-line react/jsx-key
+		<View style={ StyleSheet.absoluteFill } onLayout={ onLayout } />,
```

Comment:
Would ESLint complain if we split this into a variable and use the variable inside the array?

```jsx
const observer = <View {/* ... */ />;
return [
    observer,
    measurements,
];
```

Reply:
Actually, it shouldn’t (before refactor I had a function there and eslint was not complaining)